crates/typst-cli/src/package.rs

```diff
@@ -56,10 +59,84 @@ impl PackageStorage {
             if dir.exists() {
                 return Ok(dir);
             }
+            let parent_dir = dir.parent().unwrap();
```

Comment:
It could be cleaner to do the reverse: something like `let parent_subdir = format!("{}/{}", spec.namespace, spec.name)` at the top of the function, then `let parent_dir = cache_dir.join(&parent_subdir)` and `let dir = parent_dir.join(&spec.version)` here (and `.join(&parent_subdir).join(&spec.version)` at the previous `if let`).